### DISCUSSION TITLE
FollowTargetGoal not working

### DISCUSSION THREAD

**Discussion by Timo-Ens** [Minestom/Minestom#612] [Help]
```java
public class Zombie extends EntityCreature {
    public Zombie() {
        super(EntityType.ZOMBIE);
        this.addAIGroup(
                new EntityAIGroupBuilder()
                        .addTargetSelector(new ClosestEntityTarget(this, 15, Player.class))
                        .addGoalSelector(new FollowTargetGoal(this, Duration.of(500, ChronoUnit.MILLIS)))
                        .build()
        );
    }
}
```
Does the FollowTargetGoal not work? Or am i doing something wrong.
The Zombie won't follow me. I'm in the range of 15 blocks

**Comment by TheMode** [comment]
I have copied your code and the entity does follow me, could you provide a reproducer?

**Comment by Timo-Ens** [comment]
```java
public class Zombie extends EntityCreature {
    public Zombie() {
        super(EntityType.ZOMBIE);
        addAIGroup(
                new EntityAIGroupBuilder()
                        .addGoalSelector(new MeleeAttackGoal(this, 2, 2, ChronoUnit.SECONDS))
                        .addGoalSelector(new CustomFollowTargetGoal(this, Duration.ofMillis(500)))
                        .addTargetSelector(new ClosestEntityTarget(this, 15, Player.class))
                        .build()
        );
    }
}
```
```java

                    Zombie zombie = new Zombie();
                    zombie.setInstance(player.getInstance(), player.getPosition());
```
Also copied the FollowTargetGoal and added some logging. It seems that only the contructor get's called nothing else.
The MeleeAttackGoal works

**Reply by TheMode** [reply]
It perfectly work for me (I am using the demo server and I modified [this](https://github.com/Minestom/Minestom/blob/master/demo/src/main/java/net/minestom/demo/entity/ChickenCreature.java) entity). I will need a hello world server to reproduce. Also be sure to depend on the latest commit.

**Reply by tristan-f-r** [reply]
I'm having the same issue on the extension improvements branch.

**Reply by TheMode** [reply]
I need code to reproduce on master.

**Reply by jorisguffens** [reply]
With a lot of trial and error I figured it out.

FollowTargetGoal does not work if you use anvilloader.
```
InstanceContainer instance = instanceManager.createInstanceContainer(DimensionType.OVERWORLD);
instance.setChunkLoader(new AnvilLoader(directory.toPath()));
instanceManager.registerInstance(instance);
```

**Reply by TheMode** [reply]
`AnvilLoader` is used by default, do you mean an anvil world?

**Reply by jorisguffens** [reply]
Nevermind forget everything I said. I noticed that my custom loaded world's main surface was at Y -50, I build a higer platform above 0 and then the goal works. I guess this is a pathfinding issue then.

**Reply by emortaldev** [reply]
Possibly something to do with hydrazine

**Reply by tristan-f-r** [reply]
My issue was fixed after I updated my branch commit.
